x-pack/plugins/task_manager/server/task_claimers/strategy_mget.ts

```diff
@@ -0,0 +1,314 @@
+  const taskClaimOwnership$ = new Subject<ClaimOwnershipResult>();
+
+  claimAvailableTasks(opts)
+    .then((result) => {
+      taskClaimOwnership$.next(result);
+    })
+    .catch((err) => {
+      taskClaimOwnership$.next(err);
+    })
+    .finally(() => {
+      taskClaimOwnership$.complete();
+    });
+
+  return taskClaimOwnership$;
+}
+
+async function claimAvailableTasks(opts: TaskClaimerOpts): Promise<ClaimOwnershipResult> {
+  const { getCapacity, claimOwnershipUntil, batches, events$, taskStore } = opts;
+  const { definitions, unusedTypes, excludedTaskTypes, taskMaxAttempts } = opts;
+  const { logger } = opts;
+  const loggerTag = claimAvailableTasksMget.name;
+  const logMeta = { tags: [loggerTag] };
+  const initialCapacity = getCapacity();
+
+  const { docs, versionMap } = await searchAvailableTasks({
+    definitions,
+    excludedTaskTypes,
+    taskStore,
+    events$,
+    claimOwnershipUntil,
+    size: initialCapacity * SIZE_MULTIPLIER_FOR_TASK_FETCH,
+    taskTypes: new Set(),
+    unusedTypes,
+    taskMaxAttempts,
+  });
+
+  if (docs.length === 0) return emptyClaimOwnershipResult();
+
+  const currentTasks = new Set<ConcreteTaskInstance>();
+  const staleTasks = new Set<ConcreteTaskInstance>();
+  const missingTasks = new Set<ConcreteTaskInstance>();
+
+  const docLatestVersions = await taskStore.getDocVersions(docs.map((doc) => `task:${doc.id}`));
+
+  for (const searchDoc of docs) {
+    const searchVersion = versionMap.get(searchDoc.id);
+    const latestVersion = docLatestVersions.get(`task:${searchDoc.id}`);
+    if (!searchVersion || !latestVersion) {
+      missingTasks.add(searchDoc);
+      continue;
+    }
+
+    if (
+      searchVersion.seqNo === latestVersion.seqNo &&
+      searchVersion.primaryTerm === latestVersion.primaryTerm
+    ) {
+      currentTasks.add(searchDoc);
+      continue;
+    } else {
+      staleTasks.add(searchDoc);
+      continue;
+    }
+  }
+
+  for (const doc of missingTasks) {
+    logger.info(`Task ${doc.id} is missing from the task store or something`, logMeta);
+  }
+
+  for (const doc of staleTasks) {
+    logger.info(`Task ${doc.id} is stale`, logMeta);
+  }
+
+  const candidateTasks = applyLimitedConcurrency(currentTasks, batches);
+  const taskUpdates: ConcreteTaskInstance[] = Array.from(candidateTasks)
+    .map((task) => {
+      if (task.retryAt != null && new Date(task.retryAt).getTime() < Date.now()) {
+        task.scheduledAt = task.retryAt;
+      } else {
+        task.scheduledAt = task.runAt;
+      }
+      task.retryAt = claimOwnershipUntil;
+      task.ownerId = taskStore.taskManagerId;
+      task.status = TaskStatus.Claiming;
+
+      return task;
+    })
+    .slice(0, initialCapacity);
+
+  const finalResults: ConcreteTaskInstance[] = [];
+  let conflicts = staleTasks.size;
+  try {
+    const updateResults = await taskStore.bulkUpdate(taskUpdates, { validate: true });
```

Comment:
Similar to https://github.com/elastic/kibana/blob/main/x-pack/plugins/task_manager/server/task_running/task_runner.ts#L482 so we only validate the state when it changes and is provided by the task runner.
```suggestion
    const updateResults = await taskStore.bulkUpdate(taskUpdates, { validate: false });
```

Reply:
fixed in commit 04baedb48f6115d0f03d3d79375fc1a9e4031433